### DISCUSSION TITLE
Measuring guest kernel right before startup

### DISCUSSION THREAD

**Discussion by AlexTUM** [coconut-svsm/svsm#500] [Q&A] [answered]
Hi all,

I currently try to hack something into svsm to measure the linux guest kernel that is about to be launched. For that I need to find the best location to squeeze my code in. So that should be somewhere, where the kernel is fully loaded into memory but before the context switch to the hypervisor is prepared. In particular I need access to the guest kernel's page tables.

Any tips? (I am looking for it myself but help of experts is highly appreciated; I don't want to land on a sub-ideal result or find out something after several hours or even days of searching that somebody with a lot of insight into the code base could have told me right away)

**Comment by Freax13** [comment, marked as answer]
The SVSM doesn't currently support Linux direct boot, does it? AFAIK we only boot into regular firmware, so the place where the kernel would be loaded wouldn't be in the SVSM, but OVMF.

**Reply by AlexTUM** [reply]
Ok, that makes sense. Thanks for saving me a ton of work that would have lead to nothing 👍
